{
	"id": "oval:com.redhat.cve:def:202411168",
	"version": "636",
	"class": "vulnerability",
	"metadata": {
		"title": "python: Improper validation of IPv6 and IPvFuture addresses (moderate)",
		"reference": [
			{
				"ref_id": "CVE-2024-11168",
				"ref_url": "https://access.redhat.com/security/cve/CVE-2024-11168",
				"source": "CVE"
			}
		],
		"description": "DOCUMENTATION: A flaw was found in Python. The `urllib.parse.urlsplit()` and `urlparse()` functions improperly validated bracketed hosts (`[]`), allowing hosts that weren't IPv6 or IPvFuture compliant. This behavior was not conformant to RFC 3986 and was potentially vulnerable to server-side request forgery (SSRF) if a URL is processed by more than one URL parser.",
		"advisory": {
			"from": "secalert@redhat.com",
			"severity": "Moderate",
			"updated": {
				"date": "2024-11-25"
			},
			"cve": [
				{
					"text": "CVE-2024-11168",
					"cvss3": "3.7/CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:L/A:N",
					"cwe": "CWE-1287",
					"href": "https://access.redhat.com/security/cve/CVE-2024-11168",
					"impact": "moderate",
					"public": "20241112:2122"
				}
			],
			"affected": {
				"resolution": [
					{
						"state": "Affected",
						"component": [
							"platform-python",
							"platform-python-debug",
							"platform-python-devel",
							"python3",
							"python3-idle",
							"python3-libs",
							"python3-test",
							"python3-tkinter"
						]
					}
				]
			},
			"affected_cpe_list": {
				"cpe": [
					"cpe:/a:redhat:enterprise_linux:8",
					"cpe:/a:redhat:enterprise_linux:8::appstream",
					"cpe:/a:redhat:enterprise_linux:8::crb",
					"cpe:/a:redhat:enterprise_linux:8::highavailability",
					"cpe:/a:redhat:enterprise_linux:8::nfv",
					"cpe:/a:redhat:enterprise_linux:8::realtime",
					"cpe:/a:redhat:enterprise_linux:8::resilientstorage",
					"cpe:/a:redhat:enterprise_linux:8::sap",
					"cpe:/a:redhat:enterprise_linux:8::sap_hana",
					"cpe:/a:redhat:enterprise_linux:8::supplementary",
					"cpe:/o:redhat:enterprise_linux:8",
					"cpe:/o:redhat:enterprise_linux:8::baseos"
				]
			},
			"issued": {}
		},
		"affected": {}
	},
	"criteria": {
		"operator": "OR",
		"criterias": [
			{
				"operator": "AND",
				"criterias": [
					{
						"operator": "OR",
						"criterias": [
							{
								"operator": "AND",
								"criterions": [
									{
										"test_ref": "oval:com.redhat.cve:tst:20199674007",
										"comment": "platform-python-devel is installed"
									},
									{
										"test_ref": "oval:com.redhat.cve:tst:20199674008",
										"comment": "platform-python-devel is signed with Red Hat redhatrelease2 key"
									}
								]
							},
							{
								"operator": "AND",
								"criterions": [
									{
										"test_ref": "oval:com.redhat.cve:tst:20199674003",
										"comment": "python3-idle is installed"
									},
									{
										"test_ref": "oval:com.redhat.cve:tst:20199674004",
										"comment": "python3-idle is signed with Red Hat redhatrelease2 key"
									}
								]
							},
							{
								"operator": "AND",
								"criterions": [
									{
										"test_ref": "oval:com.redhat.cve:tst:20199674015",
										"comment": "python3-test is installed"
									},
									{
										"test_ref": "oval:com.redhat.cve:tst:20199674016",
										"comment": "python3-test is signed with Red Hat redhatrelease2 key"
									}
								]
							},
							{
								"operator": "AND",
								"criterions": [
									{
										"test_ref": "oval:com.redhat.cve:tst:20199674001",
										"comment": "platform-python is installed"
									},
									{
										"test_ref": "oval:com.redhat.cve:tst:20199674002",
										"comment": "platform-python is signed with Red Hat redhatrelease2 key"
									}
								]
							},
							{
								"operator": "AND",
								"criterions": [
									{
										"test_ref": "oval:com.redhat.cve:tst:20199674013",
										"comment": "platform-python-debug is installed"
									},
									{
										"test_ref": "oval:com.redhat.cve:tst:20199674014",
										"comment": "platform-python-debug is signed with Red Hat redhatrelease2 key"
									}
								]
							},
							{
								"operator": "AND",
								"criterions": [
									{
										"test_ref": "oval:com.redhat.cve:tst:20199674005",
										"comment": "python3 is installed"
									},
									{
										"test_ref": "oval:com.redhat.cve:tst:20199674006",
										"comment": "python3 is signed with Red Hat redhatrelease2 key"
									}
								]
							},
							{
								"operator": "AND",
								"criterions": [
									{
										"test_ref": "oval:com.redhat.cve:tst:20199674009",
										"comment": "python3-libs is installed"
									},
									{
										"test_ref": "oval:com.redhat.cve:tst:20199674010",
										"comment": "python3-libs is signed with Red Hat redhatrelease2 key"
									}
								]
							},
							{
								"operator": "AND",
								"criterions": [
									{
										"test_ref": "oval:com.redhat.cve:tst:20199674011",
										"comment": "python3-tkinter is installed"
									},
									{
										"test_ref": "oval:com.redhat.cve:tst:20199674012",
										"comment": "python3-tkinter is signed with Red Hat redhatrelease2 key"
									}
								]
							}
						]
					}
				],
				"criterions": [
					{
						"test_ref": "oval:com.redhat.cve:tst:20052541003",
						"comment": "Red Hat Enterprise Linux 8 is installed"
					}
				]
			}
		],
		"criterions": [
			{
				"test_ref": "oval:com.redhat.cve:tst:20052541004",
				"comment": "Red Hat Enterprise Linux must be installed"
			}
		]
	}
}
